src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -773,9 +783,9 @@ await foreach (var reader in ReadArrayAsync(actions))
                     continue;
                 }
 
-                values = ConvertToBlob(values, actions);
+                values = ConvertToBlob(values);
 
-                actions.RegisterForDisposal(reader);
+                _toDispose.Add(reader);
```

Comment:
how long will this live? Are we going to have to worry about running out of memory?

Reply:
until we write the  counter to destination